### ISSUE TITLE
Versions of non-arch content may differ across installations

### ISSUE THREAD

**Issue by jeanconn** [sot/skare#64]
Installation of skare 0.18 required a complicated diff and install strategy for all non-arch content (bin, build, data, doc, include, share).  What steps should be taken to avoid this for future releases?

**Comment by jeanconn** [comment]
Should flight applications be fully installed via the ska package management, as in should we call "make install" for starcheck from an application config for skare?  Or should all applications have full conda packages in the next release?

**Comment by jeanconn** [comment]
Should job updated data (as in kadi events, eng archive msid data) be in separate directories from the config and setup files?

**Comment by taldcroft** [comment]
I don't have fully-formed answers to these good questions.  I'm definitely for installing as much as possible within `arch/`, and doing everything as conda packages.  For the point of discussion, the drivers for the original architecture included:

- Make certain files easy to file and universally available in a arch-independent way.
- Put package apps into a namespace to not clutter the global path namespace.

The poster child for the first point is `task_schedule.cfg`.  But in fact as we saw this file has the potential to get out of sync with the package and that would be bad.  There is no technical reason we can't put this in as package data and have a universal app that knows how to get it, e.g. 
```
/proj/sot/ska/bin/task_schedule.pl -config `/proj/sot/ska/bin/ska_data_path Ska.engarchive task_schedule.cfg`. 
```
(Maybe `task_schedule.pl` could have some way of reducing this boilerplate)

In interactive use one might use this app like:
```
source activate ska-test
cd `ska_data_path Ska.engarchive`  # without a 2nd arg it gives the root path
# Look at data, whatever
```

One the second point, this may still be a valid concern.  At least to this point we have app names like `update_archive.py` that are pretty generic.  The problems here can largely be mitigated sticking with `share` as is, but using standard entry-point launcher scripts in all cases which call package module functions.  Updates to share would only happen when a new share app appears or disappears.

In cases were we really do want a universally available app that could be done through the standard setuptools entry point mechanism.  (One question is in development work how one can run a script that only exists after installing.  Example: xija gui_fit.py).

**Comment by jeanconn** [comment]
If we stick with `share` as is, perhaps we should require that packages/applications update a version file in their share directory to aid traceablity?

**Comment by taldcroft** [comment]
> If we stick with share as is, perhaps we should require that packages/applications update a version file in their share directory to aid traceablity?

Assuming everything in share is just a generic launcher script then a version file in share won't be of much use.  You will get the version corresponding to the package version in the particular Ska you have active.

I think an occasional `make install` is OK.  What we want here is something that mitigates the issues we had yesterday without completely re-architecting everything.  So the set of launcher apps for a task are quite stable for the most part, and simply adding a new one will never be dangerous.  These apps in share are mostly things used by cron task processing, and that may be where we draw the line between a script that goes into the arch-specific bin via setuptools vs. a static launcher in share.

**Comment by jeanconn** [comment]
> Assuming everything in share is just a generic launcher script then a version file in share 
> won't be of |much use. You will get the version corresponding to the package version in
> the particular Ska you have active.

Right, but you could programmatically check that you have run "make install" at least recently enough to get all the scripts that are required.

**Comment by taldcroft** [comment]
> Right, but you could programmatically check that you have run "make install" at least recently enough to get all the scripts that are required.

With only launcher scripts in share and no static data installed into data, then you can just always run `make install` by default without bothering with a version check, programmatic or otherwise.
